src/cuda.jl

```diff
@@ -1,28 +1,19 @@
 import .CuArrays: CuArray
 import .CuArrays.CUDAdrv: CuPtr, synchronize
+import .CuArrays.CUDAdrv.Mem: DeviceBuffer
 
-function get_ptr(buf::CuArray{T}) where T
-    _buf = Base.cconvert(CuPtr{T}, buf)
-    cuptr = Base.unsafe_convert(CuPtr{T}, _buf)
-    reinterpret(Ptr{T}, cuptr)
-end
+MPIPtr(x::CuArray{T}) where T = reinterpret(MPIPtr, Base.unsafe_convert(x))
```

Comment:
This is slightly dangerous since it loses a GC reference.

Reply:
Removed those constructors. They were not really necessary.